corehq/sql_accessors/sql_templates/get_all_cases_modified_since_1.sql

```diff
@@ -0,0 +1,17 @@
+DROP FUNCTION IF EXISTS get_all_cases_modified_since(timestamp with time zone, INTEGER, INTEGER);
+
+CREATE FUNCTION get_all_cases_modified_since(reference_date timestamp with time zone, last_id INTEGER, query_limit integer)
+RETURNS SETOF form_processor_commcarecasesql AS $$
+BEGIN
+    RETURN QUERY
+    SELECT cases.* FROM (
+        SELECT * FROM form_processor_commcarecasesql as case_table
+        WHERE case_table.server_modified_on >= reference_date
+        LIMIT query_limit + 1
```

Comment:
Yea, good question. The deal here is that we don't want to sort in the inner query because that will be very expensive (sort happens before limit). So we limit the data and then sort and re-filter to get rid of the the doc from the previous batch (which we know will be there since we're doing `>=` in the inner query).

The result is that we get `query_limit-1` docs returned for all batches except the first (and last which could have any number < query_limit).

This is similar to the approach described here: http://docs.couchdb.org/en/1.6.1/couchapp/views/pagination.html#paging-alternate-method

There is an even better way described here which requires an index on the pagination keys: https://wiki.postgresql.org/images/3/35/Pagination_Done_the_PostgreSQL_Way.pdf